scalyr_agent/agent_main.py

```diff
@@ -2183,34 +2208,33 @@ def stop(self):
     )
     my_controller.add_options(parser)
 
-    (options, args) = parser.parse_args()
-    my_controller.consume_options(options)
+    # Parse the main command by command parser.
+    command_args, other_argv = command_parser.parse_known_args()
 
-    if len(args) < 1:
-        print(
-            'You must specify a command, such as "start", "stop", or "status".',
-            file=sys.stderr,
-        )
-        parser.print_help(sys.stderr)
-        sys.exit(1)
-    elif len(args) > 1:
-        print(
-            'Too many commands specified.  Only specify one of "start", "stop", "status".',
-            file=sys.stderr,
-        )
-        parser.print_help(sys.stderr)
-        sys.exit(1)
-    elif args[0] not in (
-        "start",
-        "stop",
-        "status",
-        "restart",
-        "condrestart",
-        "version",
-    ):
-        print('Unknown command given: "%s"' % args[0], file=sys.stderr)
-        parser.print_help(sys.stderr)
-        sys.exit(1)
+    # If command is not specified, then we just print usage of the this script.
+    if command_args.command is None:
+        parser.print_help(file=sys.stderr)
+        exit(1)
+
+    if command_args.command == "service":
+        # Windows specific command that tell to start Agent's Windows service.
+        from scalyr_agent import platform_windows
+
+        # Create fully valid command line args so the Windows service could handle it properly.
+        argv = [sys.argv[0]]
+        argv.extend(other_argv)
+        platform_windows.parse_options(argv)
+        exit(0)
+
+    if command_args.command == "config":
+        # Add the config option. So we can configure the agent from the same executable.
+        # It has to save us from building multiple frozen binaries, when packaging.
+        config_main.parse_config_options(other_argv)
```

Comment:
I assume ``parse_config_options`` throws in case of errors / similar and we exit with non-zero and the line below will never be reached (`exit(1)`)?

Reply:
Yes, this function has its own argument parser that has to exit with non-zero. The line below is just to exit for sure if the parsing of the `scalyr-agent-2-config` arguments is successful.